lib/kubernetes-deploy/restart_task.rb

```diff
@@ -19,6 +20,10 @@ def initialize(deployment_name, response)
     HTTP_OK_RANGE = 200..299
     ANNOTATION = "shipit.shopify.io/restart"
 
+    def server_version
+      kubectl.server_version
```

Comment:
I don't know how to go about getting the server_version in these tasks that don't already have kubectl defined. Is it okay to require Kubectl.rb and then just define kubectl like its done in deploy_task: 
```
def kubectl
       @kubectl ||= Kubectl.new(namespace: @namespace, context: @context, logger: @logger, 
       log_failure_by_default: true)
end
```

I guess where my knowledge falls short is during run time. It looks like both restart_task and runner_task end up using deploy_task(atleast in testing) but does that mean that I can use the `server_version` function defined in deploy_task in the other classes?
@KnVerey @stefanmb

Reply:
> looks like both restart_task and runner_task end up using deploy_task(atleast in testing) 

I'm not sure what you're looking at here. The tasks share some tools (e.g. the resource watcher), but they're independent.

> Is it okay to require Kubectl.rb and then just define kubectl like its done in deploy_task

Yes. What you have now now is the correct approach IMO. I personally wouldn't bother wrapping `kubectl.server_version` as `server_version` everywhere, but that doesn't matter at all. The reason the other two tasks don't already use `kubectl` extensively is that we were trying to stick to the `Kubeclient` gem instead. But I've checked pretty carefully, and it doesn't seem to expose _any_ server metadata.